projects/batfish/src/main/java/org/batfish/z3/Z3ContextJob.java

```diff
@@ -180,20 +180,25 @@ protected BoolExpr getSolverInput(Expr answer, NodProgram program, boolean negat
     if (answer.getArgs().length > 0) {
 
       Map<String, BitVecExpr> variablesAsConsts = program.getNodContext().getVariablesAsConsts();
-      List<BitVecExpr> reversedVars =
-          Lists.reverse(
-              program
-                  .getNodContext()
-                  .getVariableNames()
-                  .stream()
-                  .filter(
-                      name ->
-                          !TransformationHeaderField.transformationHeaderFieldNames.contains(name))
-                  .map(variablesAsConsts::get)
-                  .collect(Collectors.toList()));
-
+      List<BitVecExpr> vars =
+          program
+              .getNodContext()
+              .getVariableNames()
+              .stream()
+              .filter(
+                  name -> !TransformationHeaderField.transformationHeaderFieldNames.contains(name))
+              .map(variablesAsConsts::get)
+              .collect(Collectors.toList());
+      List<BitVecExpr> reversedVars = Lists.reverse(vars);
+
+      Expr substitutedSmtConstraint =
+          program.getSmtConstraint().substituteVars(vars.toArray(new Expr[] {}));
```

Comment:
@anothermattbrown  I think we can make this more generic/useful: 

if program does not have an `smtConstraint` i.e., it defaults to the `TrueExpr.Instance`, introducing this `substitutedSmtConstraint` adds unnecessary complexity to the answer

Some of the reachability analysis I do only cares about the raw fixed point answer

Now with the line above, I get
```
(and <raw_fixed_point_answer_expression> true true)
```
So I am hacking the 'solverInput` answer I get back and extracting `solverInput.getArgs()[0]` to get my raw answer

In summary, if there is no smtconstraint, perhaps we can avoid complicating the answer

Reply:
Makes sense to me.